File: ws_p1/src/node_py/node_py/main_node.py
import rclpy
from rclpy.node import Node

class MainNode(Node):
    def __init__(self):
        super().__init__('main_node')
        self.get_logger().info('Main node started')

def main(args=None):
    rclpy.init(args=args)
    main_node = MainNode()
    rclpy.spin(main_node)
    rclpy.shutdown()

File: ws_p1/src/node_cpp/src/main_node.cpp
#include "rclcpp/rclcpp.hpp"

int main(int argc, char *argv[]) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("main_node");
    RCLCPP_INFO(node->get_logger(), "Hello world!");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}